x-pack/plugins/code/server/security.ts

```diff
@@ -66,10 +66,7 @@ export class SecureRoute {
   private isSecurityEnabledInEs() {
     // @ts-ignore
     const xpackInfo = this.server.plugins.xpack_main.info;
-    if (
-      xpackInfo.isAvailable() &&
-      (!xpackInfo.feature('security').isEnabled() || xpackInfo.license.isOneOf('basic'))
-    ) {
```

Comment:
I'll try to explain this `if` statement. It covers 2 conditions.
security disabled:
 1. user disable the security plugin on purpose(eg. by config in kibana.yml)
 2. user is using basic license

The  license type and security plugin status are not linked. I find in a basic license situation, `xpackInfo.feature('security').isEnabled() == true`